gdnative/multi-lod/src/lodmanager.cpp

```diff
@@ -170,47 +170,38 @@ void LODManager::lod_function() {
-                        lod_object_node->set("fov", fov);
-                    }
-                    // Update AABB if needed
-                    if ((update_AABBs_flag || !interacted_with_manager) && lod_object_node->get("use_screen_percentage")) {
-                        debug_level_print(2, "Telling this LOD object to update its AABBs.");
-                        lod_object_node->call("update_lod_AABB");
-                    }
-                    if (!interacted_with_manager) {
-                        debug_level_print(2, "Setting this LOD object's interaction with the manager flag to true.");
-                        lod_object_node->set("interacted_with_manager", true);
-                    }
-                    // Pass camera location and do calculations on LOD object
-                    debug_level_print(2, "Checking if the LOD object has a valid process_data function.");
-                    if (lod_object_node->has_method("process_data")) {
-                        debug_level_print(2, "It does. Calling the process_data function with the camera found.");
-                        lod_object_node->call("process_data", camera_location);
-                    } else {
-                        debug_level_print(2, "A valid process_data function was not found.");
-                    }
-                } else {
-                    debug_level_print(1, "LOD Object is NOT valid.");
+                debug_level_print(2, "LODManager: Checking if a LOD object is valid and inside the scene tree.");
+
+                if (!lod_objects[j]) {
+                    ERR_PRINT("LODManager: Invalid object found in process array.");
+                    continue;
+                }
+
+                LODBaseComponent* lbc = lod_objects[j];
+                Spatial* obj = lbc->get_object();
+
+                if (!obj->is_inside_tree()) {
+                    ERR_PRINT(obj->get_name() + String(": LOD Object is not in the scene tree"));
+                    continue;
+                }
+
+                debug_level_print(2, String("LODManager: ") + obj->get_name() + ": LOD object is valid.");
+
+                // Update multiplier if needed. It will fetch the distances from our public values
+                if (update_multipliers_flag) {
+                    debug_level_print(2, "LODManager: Telling the LOD object to update its multipliers in the LOD thread.");
+                    lbc->update_lod_multipliers_from_manager();
+                }
+
+                // Update AABB if needed
+                if (update_AABBs_flag && lbc->use_screen_percentage) {
+                    debug_level_print(2, "LODManager: Telling this LOD object to update its AABBs.");
+                    lbc->update_lod_AABB();
                 }
+
+                // Pass camera location and do calculations on LOD object
+                debug_level_print(2, "LODManager: It does. Calling the process_data function with the camera found.");
```

Comment:
Update the comment/Remove "It does" to avoid confusion (this is left over from the function validity check)